manim/mobject/types/vectorized_mobject.py

```diff
@@ -1096,6 +1260,134 @@ def get_subpaths(self) -> tuple:
         """
         return self.get_subpaths_from_points(self.points)
 
+    def get_subpath_split_indices_from_points(
+        self, points: np.ndarray, n_dims: int = 3, strip_null_end_curves: bool = False
+    ) -> np.ndarray:
+        nppcc = self.n_points_per_cubic_curve
+        starts = points[::nppcc]
+        ends = points[nppcc - 1 :: nppcc]
+        # This ensures that there are no more starts than ends.
+        # TODO: ends.shape[0] would be more efficient, but some test cases regarding
```

Comment:
I'd like to collaborate if possible!
I have many doubts and questions regarding the format of these variables, like:
- Should the `points` parameters **always have to be ndarrays**?
- Should we enforce the use of `np.asarray` or `np.array` at the beginning of most methods to ensure that we're working with NumPy arrays rather than lists, or can we sometimes just assume that without using `np.asarray`?
- What methods should be 'more exposed' to the animators creating new scenes, and thus should be prepared in case the users decide to pass a Python list instead of a NumPy array?
- etc.

No need to reply here, I think this could be better discussed in another place.